octoprint_mrbeam/static/js/wizard_analytics.js

```diff
@@ -5,16 +5,18 @@ $(function () {
         self.MY_WIZARD_TAB_NAME = "wizard_plugin_corewizard_analytics_link";
 
         self.analyticsInitialConsent = ko.observable(null);
+        self.containsAnalyticsTab = false;
 
         self.onBeforeWizardTabChange = function(next, current) {
             if (next !== self.MY_WIZARD_TAB_NAME && current === self.MY_WIZARD_TAB_NAME) {
                 let result = self._handleAnalyticsTabExit();
+                self.containsAnalyticsTab = true;
                 return result;
             }
         };
 
         self.onBeforeWizardFinish = function() {
-            if (!self.analyticsInitialConsent()) {
+            if (!self.analyticsInitialConsent() && self.containsAnalyticsTab) {
```

Comment:
If the analytics wizard is not shown, would onWizardFinish() still be called? What data would be sent to the server? I think, onWizardFinish() needs to have the self.containsAnalyticsTab as well.

Reply:
You're right. I've added it.